src/test/daml/Daml/Finance/Data/Test/DateClock.daml

```diff
@@ -0,0 +1,63 @@
+-- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Data.Test.DateClock where
+
+import DA.Assert ((===))
+import DA.Date (addDays, toDateUTC)
+import DA.Set qualified as S (singleton)
+import Daml.Finance.Data.Time.DateClock qualified as DateClock (DateClock(..), T)
+import Daml.Finance.Data.Time.DateClock.Types (Unit(..))
+import Daml.Finance.Interface.Data.Reference.Time qualified as Time (Advance(..), I, Rewind(..))
+import Daml.Finance.Interface.Types.Common.Types (Id(..))
+import Daml.Script
+
+
+testDateClock : Script ()
+testDateClock = script do
+  bank <- listKnownParties
+    >>= flip getOrAllocatePartyWithSameHint "Bank"
+  now <- getTime
+
+  let
+    dateNow = toDateUTC now
+    dateClock = DateClock.DateClock with
+      providers = S.singleton bank
+      date = Unit dateNow
+      id = Id "CLOCK"
+      description = "Clock"
+      observers = mempty
+
+  -- test advance
+  dateClockCid <- submit bank
```

Comment:
you can just use `... <- toInterfaceContractId <$> submit bank ...` - I think it's easier to read and understand

Reply:
Shwitch to  `xxx <- toInterfaceContractId @XXX <$> submit bank  do`. I find that it tends to get quite long so I was just experimenting with in other styles. Generally I find
```hs
aaa
. bbb
. ccc
. ddd
$ eee
```
easier to work with.